docs/content/en/docs/implementing/tasks/_index.md

```diff
@@ -0,0 +1,179 @@
+file.
+
+A context environment variable is available via `Deno.env.get("CONTEXT")`.
+It can be used like this:
+  
+```javascript
+let context = Deno.env.get("CONTEXT");
+    
+if (context.objectType == "Application") {
+    let application_name = contextdata.appName;
+    let application_version = contextdata.appVersion;
+}       
+        
+if (context.objectType == "Workload") {
+    let application_name = contextdata.appName;
+    let workload_name = contextdata.workloadName;
+    let workload_version = contextdata.workloadVersion;
+}
+```
+
+## Parameterized functions
+
+`KeptnTaskDefinition`s can use input parameters.
+Consider the following example:
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v1alpha2
+kind: KeptnTaskDefinition
+metadata:
+  name: slack-notification-dev
+spec:
+  function:
+    functionRef:
+      name: slack-notification
+    parameters:
+      map:
+        textMessage: "This is my configuration"
+    secureParameters:
+      secret: slack-token
+```
+
+Note the following about using parameters with functions:
+
+- The Lifecycle Toolkit passes the values
+  defined inside the `map` field as a JSON object.
+- Multi-level maps are not currently supported.
+- The JSON object can be read through the environment variable `DATA`
+  using `Deno.env.get("DATA");`.
+- Currently only one secret can be passed.
+  The secret must have a `key` called `SECURE_DATA`.
+  It can be accessed via the environment variable `Deno.env.get("SECURE_DATA")`.
+
+## Create secret text
+
+To create a secret to use in a `KeptnTaskDefinition`,
+execute this command:
+
+```yaml
+# kubectl create secret generic my-secret --from-literal=SECURE_DATA=foo
+  
+apiVersion: lifecycle.keptn.sh/v1alpha3
+kind: KeptnTaskDefinition
+metadata:
+  name: dummy-task
+  namespace: "default"
+spec: 
+  function: 
+    secureParameters:
+      secret: my-secret
+    inline:
+      code: |
+        let secret_text = Deno.env.get("SECURE_DATA");
+        // secret_text = "foo"
+```
+
+This methodology supports multiple variables
+by creating a Kubernetes secret with a JSON string:
+
+```yaml
+# kubectl create secret generic my-secret \
+# --from-literal=SECURE_DATA="{\"foo\": \"bar\", \"foo2\": \"bar2\"}"
+
+apiVersion: lifecycle.keptn.sh/v1alpha3
+kind: KeptnTaskDefinition
+metadata:
+  name: dummy-task
+  namespace: "default"
+spec:
+  function:
+    secureParameters:
+      secret: my-secret
+    inline:
+      code: |
+        let secret_text = Deno.env.get("SECURE_DATA");
+        let secret_text_obj = JSON.parse(secret_text);
+        // secret_text_obj["foo"] = "bar"
+        // secret_text_obj["foo2"] = "bar2"
+```
+
+## Pass secrets to a function
+
+In the previous example, you see that
+Kubernetes
+[secrets](https://kubernetes.io/docs/concepts/configuration/secret/)
+can be passed to the function
+using the `secureParameters` field.
+
+Here, the `secret` value is the name of the Kubernetes secret,
+which contains a field with the key `SECURE_DATA`.  
```

Comment:
We are speaking about SECRET_DATA twice in this file, is it no purpose? I mean here where the comment is and additionally here https://github.com/keptn/lifecycle-toolkit/pull/1392/files#diff-03efabffedd85e41397e0df2ecce18d41a56843483b8a37814857998ddc05559R82

and in the end also here https://github.com/keptn/lifecycle-toolkit/pull/1392/files#diff-03efabffedd85e41397e0df2ecce18d41a56843483b8a37814857998ddc05559R176

Shouldn't we mention it only once so that user won't get confused ?

Reply:
Probably.  The existing document had these two pieces so I pulled them together.  I think the whole mess needs to be rewritten -- see https://github.com/keptn/lifecycle-toolkit/issues/1390 .  I'd like to defer that until after we get the GS guides published and this PR is needed to support them.

Reply:
I also agree to have it improved in #1390